src/library/libraryview.cpp

```diff
@@ -709,6 +737,29 @@ void LibraryView::FilterReturnPressed() {
   emit doubleClicked(currentIndex());
 }
 
+void LibraryView::SearchForArtist() {
+  SaveFocus();
+  if (!last_selected_song_.albumartist().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.albumartist().simplified());
+  } else if (!last_selected_song_.artist().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.artist().simplified());
+  }
+}
+
+void LibraryView::SearchForAlbum() {
+  SaveFocus();
+  if (!last_selected_song_.album().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.album().simplified());
+  }
+}
+
+void LibraryView::SearchForThis() {
+  SaveFocus();
+  if (!last_selected_container_.isEmpty()) {
+    filter_->ShowInLibrary(last_selected_container_.simplified());
```

Comment:
So I see two options here: 
1. Search for exactly what's being displayed when a user clicks "search for this" on a divider, which might not return results if the album year is included as per the sorting preferences.
2. Modify librarymodel.cpp to record the album, album artist, or artist for each divider (depending on type of divider), and search for that. 

Which is more suitable?